transport/socks5/stream_dialer.go

```diff
@@ -23,23 +23,89 @@ import (
 	"github.com/Jigsaw-Code/outline-sdk/transport"
 )
 
+// bufferSize: The maximum buffer size is
+// 3 (1 socks version + 1 method selection + 1 methods)
+// + 1 (auth version) + 1 (username length) + 255 (username) + 1 (password length) + 255 (password)
+const (
+	maxCredentialLength = 255
+	minCredentialLength = 1
+	socksProtocolVer    = 0x05
+	noAuthMethod        = 0x00
```

Comment:
It is, however, ok to declare protocol enumerations, like the auth method, in the socks5.go file.
But they should be named with the same prefix, like authMethodNoAuth and authMethodUserPass. Keep them private, since there's no need to expose them.